analyses/doublet-detection/scripts/01b_run-scrublet.py

```diff
@@ -21,8 +21,16 @@ def run_scrublet(adata: anndata.AnnData, random_seed: int) -> pandas.DataFrame:
     """
 
     scrub = scrublet.Scrublet(adata.X, random_state = random_seed)
+
+    # Handle test datasets, which are very small
+    # The real datasets used in this module will always have > 10
+    if (scrub.n_neighbors < 10):
+        n_pc = 2
```

Comment:
How did you figure out the cutoff to use here was 10? Is there a way to specifically grab the number of pcs present in the `scrub` object and use that rather than `n_neighbors`?

Reply:
> Is there a way to specifically grab the number of pcs present in the scrub object and use that rather than n_neighbors?

I know, this 10 is less than ideal; I set it because the test datasets that were failing were below 10. I agree this should have been done in a systematic way, but I didn't see an immediate way how and didn't think it was worth the effort to dig in further. The `__init__` function for the `Scrublet` class doesn't actually perform the PCA yet - it doesn't do much of anything besides set the object up for use: https://github.com/swolock/scrublet/blob/67f8ecbad14e8e1aa9c89b43dac6638cebe38640/src/scrublet/scrublet.py#L100-L126

So, that information isn't available in the `scrub` object. I'm definitely open to other ideas though - for example, we might prefer a `try`/`except` somehow on the associated `ValueError`?

Reply:
> So, that information isn't available in the scrub object. I'm definitely open to other ideas though - for example, we might prefer a try/except somehow on the associated ValueError?

Hmm I think this could be a good approach! I think I would set it up so you run the whole thing and then if it fails, you just output a dataframe with barcodes and `NA` for the scrublet results columns rather than guessing at the best `n_pcs` to use.

Reply:
better parallel with `scDblFinder` too!

Reply:
Ah, this will mean I need to further update notebooks to handle cases where scrublet results are `NA`. Hopefully that's not too involved but if it is I'll let you know...

Reply:
Not too involved! ✅